===================================================================
RCS file: /cvs/gnome/nautilus/src/nautilus-places-sidebar.c,v
retrieving revision 1.17
retrieving revision 1.18
diff -u -r1.17 -r1.18
--- nautilus-places-sidebar.c	2006/04/07 00:17:11	1.17
+++ src/nautilus-places-sidebar.c	2006/04/15 16:19:44	1.18
@@ -961,6 +961,7 @@
 {
 	GtkTreeIter iter;
 	GtkTreeSelection *selection;
+	PlaceType type; 
 	int index;
 
 	selection = gtk_tree_view_get_selection (sidebar->tree_view);
@@ -970,6 +971,14 @@
 	}
 	
 	gtk_tree_model_get (GTK_TREE_MODEL (sidebar->store), &iter,
+			    PLACES_SIDEBAR_COLUMN_ROW_TYPE, &type,
+			    -1);
+
+	if (type != PLACES_BOOKMARK) {
+		return;
+	}
+
+	gtk_tree_model_get (GTK_TREE_MODEL (sidebar->store), &iter,
 			    PLACES_SIDEBAR_COLUMN_INDEX, &index,
 			    -1);
 
